src/modules/workspace/promote.ts

```diff
@@ -1,45 +1,108 @@
 import chalk from 'chalk'
+import axios from 'axios'
+import { formatHyperlink } from '../utils'
+import boxen from 'boxen'
+import ora from 'ora'
 import { CommandError } from '../../api/error/errors'
 import { createWorkspacesClient } from '../../api/clients/IOClients/infra/Workspaces'
 import { SessionManager } from '../../api/session/SessionManager'
 import log from '../../api/logger'
 import { promptConfirm } from '../../api/modules/prompts'
+import { VBase } from '../../api/clients/IOClients/infra/VBase'
+import authUrl from '../url'
 import useCmd from './use'
+import { ColorifyConstants } from '../../lib/constants/Colors'
 
+const { checkForConflicts } = VBase.createClient()
 const { promote, get } = createWorkspacesClient()
 const { account, workspace: currentWorkspace } = SessionManager.getSingleton()
+const workspaceUrl = authUrl()
 
 const throwIfIsMaster = (workspace: string) => {
   if (workspace === 'master') {
     throw new CommandError(`It is not possible to promote workspace ${workspace} to master`)
   }
 }
 
+const handleConflict = async () => {
+  const conflictsFound = await checkForConflicts()
+  if (conflictsFound) {
+    await axios.get(workspaceUrl)
+  }
+}
+
 const isPromotable = async (workspace: string) => {
   throwIfIsMaster(workspace)
+
   const meta = await get(account, currentWorkspace)
   if (!meta.production) {
     throw new CommandError(
-      `Workspace ${chalk.green(currentWorkspace)} is not a ${chalk.green(
+      `Workspace ${ColorifyConstants.ID(workspace)} is not a ${ColorifyConstants.ID(
         'production'
-      )} workspace\nOnly production workspaces may be promoted\nUse the command ${chalk.blue(
+      )} workspace\nOnly production workspaces may be promoted\nUse the command ${ColorifyConstants.COMMAND_OR_VTEX_REF(
         'vtex workspace create <workspace> --production'
       )} to create a production workspace`
     )
   }
+
+  const spinner = ora('Preparing the workspace to be promoted').start()
+  spinner.color = 'magenta'
```

Comment:
Could this color also be a constant?

Reply:
Sure! ora.Color only accepts some specific strings, not hex code like I was doing, but I'll add `magenta` in the constants file 😄